packages/integrations/node/test/headers.test.js

```diff
@@ -7,124 +7,169 @@ describe('Node Adapter Headers', () => {
+				'set-cookie': ['from1=astro1', 'from2=astro2'],
+			});
 		});
-	});
 
-	it('Component Response Single Cookie Header', async () => {
-		await runTest('/astro/component-response-cookies-single', {
-			'content-type': 'text/html',
-			'set-cookie': 'from1=value1',
+		it('Component Response Single Cookie Header', async () => {
+			await runTest('/astro/component-response-cookies-single', {
+				'content-type': 'text/html',
+				'set-cookie': 'from1=value1',
+			});
 		});
-	});
 
-	it('Component Response Multi Cookie Header', async () => {
-		await runTest('/astro/component-response-cookies-multi', {
-			'content-type': 'text/html',
-			'set-cookie': ['from1=value1', 'from2=value2'],
+		it('Component Response Multi Cookie Header', async () => {
+			await runTest('/astro/component-response-cookies-multi', {
+				'content-type': 'text/html',
+				'set-cookie': ['from1=value1', 'from2=value2'],
+			});
 		});
-	});
 
-	it('Component Astro and Response Single Cookie Header', async () => {
-		await runTest('/astro/component-astro-response-cookie-single', {
-			'content-type': 'text/html',
-			'set-cookie': ['from1=response1', 'from1=astro1'],
+		it('Component Astro and Response Single Cookie Header', async () => {
+			await runTest('/astro/component-astro-response-cookie-single', {
+				'content-type': 'text/html',
+				'set-cookie': ['from1=response1', 'from1=astro1'],
+			});
+		});
+
+		it('Component Astro and Response Multi Cookie Header', async () => {
+			await runTest('/astro/component-astro-response-cookie-multi', {
+				'content-type': 'text/html',
+				'set-cookie': ['from1=response1', 'from2=response2', 'from3=astro1', 'from4=astro2'],
+			});
+		});
+
+		it('sends several chunks', async () => {
+			const { handler } = await import('./fixtures/headers/dist/server/entry.mjs');
+
+			const { req, res, done } = createRequestAndResponse({
+				method: 'GET',
+				url: '/astro/component-simple',
+			});
+
+			handler(req, res);
+
+			req.send();
+
+			const chunks = await done;
+			assert.equal(chunks.length, 3);
 		});
 	});
 
-	it('Component Astro and Response Multi Cookie Header', async () => {
-		await runTest('/astro/component-astro-response-cookie-multi', {
-			'content-type': 'text/html',
-			'set-cookie': ['from1=response1', 'from2=response2', 'from3=astro1', 'from4=astro2'],
+	describe('without streaming', () => {
+		before(async () => {
+			fixture = await loadFixture({
+				root: './fixtures/headers/',
+				output: 'server',
+				adapter: nodejs({ mode: 'middleware', experimentalDisableStreaming: true }),
+			});
+			await fixture.build();
+		});
+
+		it('sends a single chunk', async () => {
+			const { handler } = await import('./fixtures/headers/dist/server/entry.mjs?cachebust=0');
+
+			const { req, res, done } = createRequestAndResponse({
+				method: 'GET',
+				url: '/astro/component-simple',
+			});
+
+			handler(req, res);
+
+			req.send();
+
+			const chunks = await done;
+			assert.equal(chunks.length, 1);
```

Comment:
Can you also check that it sets content-length and doesn't set transfer-encoding to chunked?

Reply:
I don't know, in both cases they're passed the app instance with the right streaming setting

Reply:
I checked and we can't check the headers with the current testing setup